File: android/lib-ezlog/src/main/java/wtf/s1/ezlog/EZLogger.java
package wtf.s1.ezlog;

import org.jetbrains.annotations.NotNull;

import static wtf.s1.ezlog.EZLog.VERBOSE;
import static wtf.s1.ezlog.EZLog.DEBUG;
import static wtf.s1.ezlog.EZLog.INFO;
import static wtf.s1.ezlog.EZLog.WARN;
import static wtf.s1.ezlog.EZLog.ERROR;

public class EZLogger {

    private @NotNull final String loggerName;

    public EZLogger(EZLogConfig config) {
        this.loggerName = config.logName;
        EZLog._createLogger(config);
    }

    public void v(String tag, String msg) {
        EZLog._log(loggerName, VERBOSE, tag, msg);
    }

    public void d(String tag, String msg) {
        EZLog._log(loggerName, DEBUG, tag, msg);
    }

    public void i(String tag, String msg) {
        EZLog._log(loggerName, INFO, tag, msg);
    }

    public void w(String tag, String msg) {
        EZLog._log(loggerName, WARN, tag, msg);
    }

    public void e(String tag, String msg) {
        EZLog._log(loggerName, ERROR, tag, msg);
    }


}

File: android/lib-ezlog/src/main/java/wtf/s1/ezlog/EZLogger.kt
package wtf.s1.ezlog

import wtf.s1.ezlog.EZLog.DEBUG
import wtf.s1.ezlog.EZLog.ERROR
import wtf.s1.ezlog.EZLog.INFO
import wtf.s1.ezlog.EZLog.VERBOSE
import wtf.s1.ezlog.EZLog.WARN
import wtf.s1.ezlog.EZLog.createLogger
import wtf.s1.ezlog.EZLog.flush
import wtf.s1.ezlog.EZLog.log

class EZLogger(config: EZLogConfig) {
    private val loggerName: String

    init {
        loggerName = config.logName
        createLogger(config)
    }

    fun v(tag: String, msg: String) {
        log(loggerName, VERBOSE, tag, msg)
    }

    fun d(tag: String, msg: String) {
        log(loggerName, DEBUG, tag, msg)
    }

    fun i(tag: String, msg: String) {
        log(loggerName, INFO, tag, msg)
    }

    fun w(tag: String, msg: String) {
        log(loggerName, WARN, tag, msg)
    }

    fun e(tag: String, msg: String) {
        log(loggerName, ERROR, tag, msg)
    }

    fun flush() {
        flush(loggerName)
    }
}